src/main/java/frc/robot/commands/FeederIndexCommand.java

```diff
@@ -5,53 +5,68 @@
 
 public class FeederIndexCommand extends CommandBase {
     private boolean isBallBetweenSensors; // ball state
+    private boolean isRobotEmpty;
     
     public FeederIndexCommand() {
         addRequirements(Robot.FEEDER_SUBSYSTEM);
     }
 
-        // Called when the command is initially scheduled.
+    // Called when the command is initially scheduled.
     @Override
     public void initialize() {}
 
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
-        System.out.println(Robot.CONVEYANCE_SUBSYSTEM.getConveyanceOneSubsystemHasBall());
-        System.out.println(Robot.FEEDER_SUBSYSTEM.getFeederSubsystemHasBall());
-
-        // (False means no ball in front of the sensor and vice versa)
-        
-        // If there is a ball in conveyance stage 1 and no ball in conveyance stage 2 (feeder subsystem),
-        // run the conveyance stage 2 and set isBallBetweenSensors to true
-        if (Robot.CONVEYANCE_SUBSYSTEM.getConveyanceOneSubsystemHasBall() == true && Robot.FEEDER_SUBSYSTEM.getFeederSubsystemHasBall() == false) {
-            Robot.FEEDER_SUBSYSTEM.setConveyanceNormalSpeedForward();
+        // False means no ball in front of the sensor and vice versa
+        // For current testing purposes, pressing the button changes the value to false and vice versa
+        // (i.e. holding both buttons means the robot is empty 
+        // and leaving both buttons untouched means the robot is holding both balls)
+
+        // If there are any number of balls in the robot, in any conveyance stage
+        // set isRobotEmpty to false
+        if(Robot.CONVEYANCE_SUBSYSTEM.getConveyanceOneSubsystemHasBall() == true || Robot.FEEDER_SUBSYSTEM.getFeederSubsystemHasBall() == true || isBallBetweenSensors == true) {
+            isRobotEmpty = false;
+        }
+        // If there is a ball in conveyance stage 1, no ball in conveyance stage 2 (feeder subsystem), and the robot is not empty
+        // set isBallBetweenSensors to true
+        if (Robot.CONVEYANCE_SUBSYSTEM.getConveyanceOneSubsystemHasBall() == true && Robot.FEEDER_SUBSYSTEM.getFeederSubsystemHasBall() == false && isRobotEmpty == false) {
```

Comment:
Use "else if" here, its easier to reason that the first matching condition is what is acted on. In this case though, we act on every matching condition.